ui/src/core/index.ts

```diff
@@ -506,6 +506,36 @@ export function generateCore() {
 		return ca;
 	}
 
+	/**
+	 * Gets children of a specific Streamsync component.
+	 *
+	 * @param parentId The parent id.
+	 * @param includeBMC Whether to include Builder-managed components
+	 * @param includeCMC Whether to include code-managed components
+	 * @returns An array of components.
+	 */
+	function getComponentChildren(
+		parentId: Component["id"],
+		includeBMC: boolean = true,
+		includeCMC: boolean = true,
+	): Component[] {
```

Comment:
suggestion: I think having `getComponents` and `getComponentChildren` will be confusing. Maybe it would be a good idea to merge those 2 functions into one? 
```suggestion
	function getComponents(
		parentId: Component["id"],
		{
			includeBMC = true,
			includeCMC = true,
			sortedByPosition = false,
		}: {
			includeBMC?: boolean;
			includeCMC?: boolean;
			sortedByPosition?: boolean;
		} = {},
	): Component[] {
```

Reply:
I liked your "options" object argument. I'll go with that.

My idea was to leave `getComponents` when we need to get all components, and `getComponentChildren` when we need the children of a component.

I'll go for the proposed syntax and unify under `getComponents`. `parentId` needs to be optional though, as sometimes we get everything.